cloud/edgecontroller/pkg/controller/manager/location_test.go

```diff
@@ -0,0 +1,161 @@
+	}{
+		{
+			name: "TestAddOrUpdatePod(): Case 1: LocationCache is empty",
+			lc:   &LocationCache{},
+			pod:  pod,
+		},
+		{
+			name: "TestAddOrUpdatePod(): Case 2: LocationCache is not empty",
+			lc:   &locationCache,
+			pod:  pod,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			test.lc.AddOrUpdatePod(test.pod)
+		})
+	}
+}
+
+//TestConfigMapNodes is function to test ConfigMapNodes
+func TestConfigMapNodes(t *testing.T) {
+	locationCache := LocationCache{}
+	locationCache.configMapNode.Store("ObjectMeta1/VolumeConfig1", nodes)
+	tests := []struct {
+		name          string
+		lc            *LocationCache
+		namespace     string
+		configMapName string
+		nodes         []string
+	}{
+		{
+			name:  "TestConfigMapNodes(): Case 1: LocationCache is empty",
+			lc:    &LocationCache{},
+			nodes: nil,
+		},
+		{
+			name:          "TestConfigMapNodes(): Case 2: LocationCache is not empty",
+			lc:            &locationCache,
+			namespace:     "ObjectMeta1",
+			configMapName: "VolumeConfig1",
+			nodes:         nodes,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if nodes := test.lc.ConfigMapNodes(test.namespace, test.configMapName); !reflect.DeepEqual(nodes, test.nodes) {
+				t.Errorf("Manager.TestConfigMapNodes() case failed: got = %v, Want = %v", nodes, test.nodes)
+			}
+		})
+	}
+}
+
+//TestSecretNodes is function to test SecretNodes
+func TestSecretNodes(t *testing.T) {
+	locationCache := LocationCache{}
+	locationCache.secretNode.Store("ObjectMeta1/VolumeSecret1", nodes)
+	tests := []struct {
+		name       string
+		lc         *LocationCache
+		namespace  string
+		secretName string
+		nodes      []string
+	}{
+		{
+			name:  "TestSecretNodes(): Case 1: LocationCache is empty",
+			lc:    &LocationCache{},
+			nodes: nil,
+		},
+		{
+			name:       "TestSecretNodes(): Case 2: LocationCache is not empty",
+			lc:         &locationCache,
+			namespace:  "ObjectMeta1",
+			secretName: "VolumeSecret1",
+			nodes:      nodes,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if nodes := test.lc.SecretNodes(test.namespace, test.secretName); !reflect.DeepEqual(nodes, test.nodes) {
+				t.Errorf("Manager.TestSecretNodes() case failed: got = %v, Want = %v", nodes, test.nodes)
+			}
+		})
+	}
+}
+
+//TestDeleteConfigMap is function to test DeleteConfigMap
+func TestDeleteConfigMap(t *testing.T) {
+	locationCache := LocationCache{}
+	locationCache.configMapNode.Store("ObjectMeta1/VolumeConfig1", nodes)
+	namespace := "ObjectMeta1"
+	configMapName := "VolumeConfig1"
+	locationCache.DeleteConfigMap(namespace, configMapName)
+}
+
+//TestDeleteSecret is function to test DeleteSecret
+func TestDeleteSecret(t *testing.T) {
+	locationCache := LocationCache{}
+	locationCache.secretNode.Store("ObjectMeta1/VolumeSecret1", nodes)
+	namespace := "ObjectMeta1"
+	secretName := "VolumeSecret1"
+	locationCache.DeleteSecret(namespace, secretName)
```

Comment:
we should do a test to check if the secret is deleted by doing a get from the cache, it should return false / nil.